jwst/cube_build/src/cube_match_sky.c

```diff
@@ -0,0 +1,1441 @@
+    
+    Paramteter
+    ---------
+    overlap_partial - intermediate DQ flag
+    coord1: xi coordinates of input data (~x coordinate in IFU space)
+    coord2: eta coordinates of input data (~y coordinate in IFU space)
+    wave: wavelength of input data
+    roiw_ave: average spectral roi used to determine which wavelength bins
+    the input values would be mapped to
+    slice_no: integer slice value of input data (used in MIRI case to find
+    the points of the edge slices.)
+
+  */
+  
+
+  int *idqv = NULL;  // int vector for spaxel
+
+  if (mem_alloc_dq(ncube, &idqv)) return 1;
+
+  // Set all data to zero
+  for (long i = 0; i < ncube; i++){
+    idqv[i] = 0;
+  }
+
+  //  for each of the 30 slices - find the projection of this slice
+  //     onto each of the IFU wavelength planes.
+
+  for (int w = 0; w  < nz; w++) {
+    int n_slice_found = 0;
+    for (int islice =1; islice< 31 ; islice++){
+      double c1_min;
+      double c2_min;
+      double c1_max;
+      double c2_max;
+      int status = 0; 
+      status =  slice_wave_plane_nirspec( w, islice, roiw_ave, zc,
+				   coord1, coord2, wave, sliceno, ncube, npt,
+				   &c1_min, &c2_min, &c1_max, &c2_max);
+	
+      if( status ==0){
+	
+	n_slice_found++;
+	int nxy = nx * ny;
+	int wave_slice_dq[nxy];
+	for (int j =0; j< nxy; j++){
+	  wave_slice_dq[j] = 0;
+	}
+	status = overlap_slice_with_spaxels(overlap_partial,
+					    cdelt1,cdelt2,
+					    nx, ny,
+					    xc, yc,
+					    c1_min, c2_min, c1_max, c2_max,
+					    wave_slice_dq);
+      
+	long istart = nxy*w;
+	long iend = istart + nxy;
+	for( long in = istart; in < iend; in ++){
+	  long ii = in - istart;
+	  idqv[in] = wave_slice_dq[ii];
+	}
+      } // end loop over status
+      
+    } // end loop over slices
+	
+  } // end of wavelength
+  *spaxel_dq = idqv;
+
+  return 0;
+}
+
+// Match point cloud to sky and determine the weighting to assign to each point cloud  member
+// to matched spaxel based on ROI - weighting type - emsm
+
+int match_point_emsm(double *xc, double *yc, double *zc,
+		     double *coord1, double *coord2, double *wave,
+		     double *flux, double *err,
+		     double *rois_pixel, double *roiw_pixel, double *scalerad_pixel,
+		     double *zcdelt3,
+		     int nx, int ny, int nwave, int ncube, int npt,
+		     double cdelt1, double cdelt2,
+		     double **spaxel_flux, double **spaxel_weight, double **spaxel_var,
+		     double **spaxel_iflux) {
+
+
+  /*
+return values: spaxel_flux, spaxel_weight, spaxel_var, spaxel_iflux
+*/
+    double *fluxv = NULL, *weightv=NULL, *varv=NULL ;  // vectors for spaxel 
+    double *ifluxv = NULL;  // vector for spaxel
+
+    // allocate memory to hold output 
+    if (mem_alloc(ncube, &fluxv, &weightv, &varv, &ifluxv)) return 1;
+    
+    double set_zero=0.0;
+    // Set all data to zero
+    for (int i = 0; i < ncube; i++){
+      varv[i] = set_zero;
+      fluxv[i] = set_zero;
+      ifluxv[i] = set_zero;
+      weightv[i] = set_zero;
+    }
```

Comment:
This can be simplified as in `cube_match_internal.c` if using `alloc_flux_arrays`.

Reply:
good suggestion. I moved alloc_flux_arrays to cube_utils.c